### ISSUE TITLE
Crash on first try with sub folders

### ISSUE THREAD

**Issue by Voljega** [retro-esp32/RetroESP32#7]
Just downloaded firmware + skeleton sd card and burned the firmware

On first boot I chose my theme color and then went on NES system at the top.
I was told we could use subfolders in this firmware, so I just copied my setup from a retrogaming distrib for the four main system :

![retro32](https://user-images.githubusercontent.com/1405681/61562096-ccc48100-aa70-11e9-95f3-09f5daf903e1.png)

With this subfolders, no games are displayed for one system, just one system has one game displayed. But Each system contains the whole romset in these subfolders ...
like this :

![retro32-nes-action](https://user-images.githubusercontent.com/1405681/61562224-2036cf00-aa71-11e9-9362-e608288160a0.png)

Anyway nes was saying 'no games found' I clicked A anyway, then something like 'Run' appeared, I clicked A again and the emulator seemed to launch, but I ended up with the white screen and red crossed sd card icon, the one saying ' - Files are not properly written to the SD card, or its contents are corrupted.' (https://wiki.odroid.com/odroid_go/emulator/usage_go_play)

And now I can't do anything, Force exit emulator (menu hold when powering on) does nothing, nor menu, nor reset, seems the firmware is totally stuck

so TLDR :
- subfolders doesn't to be supported at all
- when the emulator crashes there doesn't seem to be anyway to go back to menu
- command run in system with no game when clicking on it

**Comment by 32teeth** [comment]
I like this feature alot.
We will implement in next few days

**Comment by 32teeth** [comment]
Closing this one and creating two separate Issues

- [ ] [Ability to Navigate Sub Folders in ROM Directories](https://github.com/retro-esp32/RetroESP32/issues/8)
- [ ] [Emulator Crashes if Target File is Blank](https://github.com/retro-esp32/RetroESP32/issues/9)

**Comment by Voljega** [comment]
OK but in the meantime is there anyway to come back to menu or will I need to reflash the firmware ?

**Comment by 32teeth** [comment]
All emulators have a Wake Up interrupt

```
            if (bootState.values[ODROID_INPUT_MENU])
            {
                // Force return to menu to recover from
                // ROM loading crashes

                // Set menu application
                odroid_system_application_set(0);

                // Reset
                esp_restart();
            }
```

You can power up and hold **Menu** (Start + Select) to go back to Retro ESP, after a few seconds you will see the Retro ESP Restarting... Screen

![320x240](https://user-images.githubusercontent.com/75797/61565149-2794b300-aa46-11e9-9a52-8476a4bd61e8.gif)

Or...  ...power up and hold **B** to go back to Odroid Firmware (Unless you used ```./ota``` script)

**Comment by Voljega** [comment]
OK it worked with start + select

**Comment by AmitMamoria** [comment]
Hi can someone please tell me how to run retro arch launcher or how to run games from sd card. I am willing to make my own retro console from scratch ie. using with individual parts  instead of buying pre-made boards that are available on google.
I have googled and found some requirements for making retro console. 

I have Following parts with me.
1.) Esp32 Dev Kit Board ( https://www.electronicscomp.com/esp32-development-board-with-wifi-bluetooth-india?gclid=CjwKCAjw74b7BRA_EiwAF8yHFE4jsQzOF9uV9N4Jpo5NEkk4oYJbGYL4lAK6atGl-U6_IuWaywzbmxoCOF0QAvD_BwE)
2.) TFT LCD
3.) SD Card Reader.

 Please Help !!

**Comment by fizarum** [comment]
Hi, first of all you can check this repo: https://github.com/airainchoc/RetroESP32/tree/Hardware and this BOM: https://github.com/airainchoc/RetroESP32/blob/Hardware/bom/RetroESP32.csv
make sure your board has this esp32 module: ESP32-WROVER (16MB) there is some picture of this soc: https://ae01.alicdn.com/kf/Hf494a4a947114e01b17cf46851904a443.jpg_q50.jpg
so it looks different from Esp32 Dev Kit Board
